### ISSUE TITLE
Refactor Storage module for Substrate

### ISSUE THREAD

**Issue by siman** [Joystream/joystream#32]

**Comment by jfinkhaeuser** [comment]
Consider that replacing all the deprecated sections with https://github.com/Joystream/substrate-runtime-joystream/blob/development/src/storage/data_object_storage_registry.rs#L73 means that the claims a storage node can make on chain change.

At that point, they can only claim that they can serve content. Before, they can claim that they intend to serve content, and then later claim that that's working.

Consider what this means for a liaison in this scenario:

1. Client creates data object, gets a liaison.
1. Client starts upload to liaison.
1. Client aborts upload.

All the chain records at this point is that the chosen liaison isn't serving the content. It's impossible to know whether this is the liaison's fault, or as described the client's. Here's a worse version of the same:

1. Client creates data object, gets a liaison.
1. Nothing more...

Consider the old/current model instead:

1. Client creates data object, gets a liaison.
1. Client starts upload to liaison.
1. Liaison creates DOSR(ready = false).
1. Client aborts upload.

We have a record that the liaison (probably; there's no counter-signing by the client) was contacted and tried its thing. We still don't know for sure what happened (bug in the liaison? client aborted?), but we know a little more.

I suspect that in the long term, we want a counter-signing mechanism similar to the role staking here.

TL;DR is, I'm actually inclined to leave this as is for the time being.

**Comment by mnaamani** [comment]
Perhaps merge this https://github.com/Joystream/substrate-runtime-joystream/pull/37 before refactoring
